src/Compilers/CSharp/Portable/Emitter/EditAndContinue/CSharpSymbolMatcher.cs

```diff
@@ -662,6 +679,49 @@ internal bool TryFindAnonymousType(AnonymousTypeManager.AnonymousTypeTemplateSym
                 return _anonymousTypeMap.TryGetValue(type.GetAnonymousTypeKey(), out otherType);
             }
 
+            internal bool TryFindAnonymousDelegate(AnonymousTypeManager.AnonymousDelegateTemplateSymbol type, out AnonymousTypeValue otherType)
+            {
+                Debug.Assert((object)type.ContainingSymbol == (object)_sourceAssembly.GlobalNamespace);
+
+                // Anonymous delegates are indexed by name, and the names are <>f__AnonymousDelegate0, 1, ... .
+                // Within a compilation, the index in the name is determined by source order, but the actual signature
+                // of the delegate type may change between generations. For instance, the signature of a lambda
+                // expression may have been changed explicitly, or another lambda expression may have been inserted
+                // ahead of this one in source order. Therefore, we need to verify the signatures of the delegate types
+                // are equivalent - by comparing the Invoke() method signatures.
+                if (_anonymousDelegateMap.TryGetValue(type.Name, out otherType) &&
```

Comment:
>type.Name [](http://example.com/codeflow?start=54&length=9)

Why use name as the key if it is not sufficient to find the match?

Reply:
The alternative would be use the signature of the delegate as the key, but since we're mapping types between generations, we'd need to map the parameter types and return type of the delegate type `type` to construct the signature key, and that mapping is equivalent to what `isCorrespondingAnonymousDelegate()` is doing below.